### PR TITLE
Leaning more on the alive (vs delete) semantics.

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: common/src/bitset.rs [hunk 1]
```diff
@@ -369,7 +366,11 @@ impl ReadSerializedBitSet {
         b & (1u8 << shift) != 0
     }
 
-    /// Returns the max_value.
+    /// Maximum value the bitset may contain.
+    /// (Note this is not the maximum value contained in the set.)
+    ///
+    /// A bitset has an intrinsic capacity.
+    /// It only stores elements within [0..max_value).
```
Comment (PSeitz) [inline common/src/bitset.rs:373]: this comment suggests to maybe rename it to capacity

File: src/core/segment_reader.rs [hunk 2]
```diff
@@ -69,17 +69,9 @@ impl SegmentReader {
         &self.schema
     }
 
-    /// Return the number of documents that have been
-    /// deleted in the segment.
-    pub fn num_deleted_docs(&self) -> DocId {
```
Comment (PSeitz) [inline src/core/segment_reader.rs:74]: Isn't this a public API someone may use currently?

**Review by PSeitz — APPROVED** [review]
LGTM, except maybe avoid API change

File: src/core/segment_reader.rs (hunk 2 above)
Comment (fulmicoton) [reply src/core/segment_reader.rs:74]: That's true. yes let's readd this.

File: common/src/bitset.rs (hunk 1 above)
Comment (fulmicoton) [reply common/src/bitset.rs:373]: Yeah it does feel weird right? Capacity is not great either though.

File: src/core/segment_reader.rs (hunk 2 above)
Comment (PSeitz) [reply src/core/segment_reader.rs:74]: Ah we still need this method in demux

File: common/src/bitset.rs (hunk 1 above)
Comment (PSeitz) [reply common/src/bitset.rs:373]: yes I agree

Comment (shikhar) [reply common/src/bitset.rs:373]: maybe limit?

Comment (fulmicoton) [reply common/src/bitset.rs:373]: `num_bits`, `num_capacity_bits` ?

Comment (shikhar) [reply common/src/bitset.rs:373]: `width`!

Comment (fulmicoton) [reply common/src/bitset.rs:373]: `bit_width` is quite good actually.

**Pull request merged**